migrations/2025-07-22-091706_multiple_emails/up.sql

```diff
@@ -0,0 +1,127 @@
+-- Drop the unique constraint on user_id to allow multiple emails per user
+ALTER TABLE emails DROP CONSTRAINT emails_user_id_key;
+
+-- Limit users to 32 emails maximum
+CREATE FUNCTION enforce_max_emails_per_user()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF (SELECT COUNT(*) FROM emails WHERE user_id = NEW.user_id) > 32 THEN
+    RAISE EXCEPTION 'User cannot have more than 32 emails';
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER trigger_enforce_max_emails_per_user
+BEFORE INSERT ON emails
+FOR EACH ROW
+EXECUTE FUNCTION enforce_max_emails_per_user();
+
+-- Add a unique constraint for the combination of user_id and email
+ALTER TABLE emails ADD CONSTRAINT unique_user_email UNIQUE (user_id, email);
+
+-- Add a new column for identifying if an email should receive notifications
+ALTER TABLE emails ADD COLUMN send_notifications BOOLEAN DEFAULT FALSE NOT NULL;
```

Comment:
I'm wondering about the `send_notifications` naming, since it could be confused with the "please send me publish notifications" flag on the users table. GitHub is using "primary" naming, which might be harder to confuse 🤔